datasets/monash_tsf/monash_tsf.py

```diff
@@ -0,0 +1,566 @@
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Monash Time Series Forecasting Repository Dataset."""
+
+
+from dataclasses import dataclass
+from datetime import datetime
+from pathlib import Path
+from typing import List, Optional
+
+import numpy as np
+from pandas.tseries.frequencies import to_offset
+
+import datasets
+
+from .utils import convert_tsf_to_dataframe, frequency_converter
+
+
+# Find for instance the citation on arxiv or on the dataset repo/website
+_CITATION = """\
+@InProceedings{godahewa2021monash,
+    author = "Godahewa, Rakshitha and Bergmeir, Christoph and Webb, Geoffrey I. and Hyndman, Rob J. and Montero-Manso, Pablo",
+    title = "Monash Time Series Forecasting Archive",
+    booktitle = "Neural Information Processing Systems Track on Datasets and Benchmarks",
+    year = "2021",
+    note = "forthcoming"
+}
+"""
+
+_DESCRIPTION = """\
+Monash Time Series Forecasting Repository which contains 30+ datasets of related time series for global forecasting research. This repository includes both real-world and competition time series datasets covering varied domains.
+"""
+
+_HOMEPAGE = "https://forecastingdata.org/"
+
+_LICENSE = "The Creative Commons Attribution 4.0 International License. https://creativecommons.org/licenses/by/4.0/"
+
+_ROOT_URL = "https://zenodo.org/record"
+
+
+@dataclass
+class MonashTSFBuilderConfig(datasets.BuilderConfig):
+    """MonashTSF builder config with some added meta data."""
+
+    file_name: Optional[str] = None
+    record: Optional[str] = None
+    prediction_length: Optional[int] = None
+    item_id_column: Optional[str] = None
+    data_column: Optional[str] = None
+    target_fields: Optional[List[str]] = None
+    feat_dynamic_real_fields: Optional[List[str]] = None
+    multivariate: bool = False
+    rolling_evaluations: int = 1
```

Comment:
Awesome ! This is nice to see that it has the same `prediction_length` and `rolling_evaluations` as in `electricity_load_diagrams`